src/NHibernate/Engine/ISessionImplementor.cs

```diff
@@ -258,6 +258,11 @@ public partial interface ISessionImplementor
 		FlushMode FlushMode { get; set; }
 
 		string FetchProfile { get; set; }
+		
+		/// <summary>
+		/// Determines whether <see cref="SessionId"/> is tracked by <see cref="SessionIdLoggingContext"/> or not.
+		/// </summary>
+		bool IdTracking { get; }
```

Comment:
I was not seeing that this way. `SessionId` only serves for logging, just set it to `Guid.Empty` in the session itself when the option disable it. No massive change. No potential breaking change by adding an interface member (although probably very few implements their own `ISessionImplementor`).

Reply:
My bad, I didn't read your comment good enough. I made an [another version](https://github.com/nhibernate/nhibernate-core/pull/1394/commits/540eb1813e0df36e6b232dcef69481d2b0607be1) that takes into account your suggestions.